cargo-dist/src/backend/installer/docker.rs

```diff
@@ -0,0 +1,203 @@
+    pub target: String,
+    /// Release this is for
+    pub release_idx: ReleaseIdx,
+    /// What to tag the image as
+    pub tag: String,
+    /// Binaries we'll be baking into the docker image
+    pub bins: Vec<String>,
+    /// Final file path of the docker image
+    pub file_path: Utf8PathBuf,
+    /// Dir stuff goes to
+    pub package_dir: Utf8PathBuf,
+    /// Additional dirs/files to include
+    pub includes: Vec<Utf8PathBuf>,
+}
+
+/// Info about running cargo-dist in Github CI
+#[derive(Debug, Serialize)]
+pub struct DockerfileInfo {
+    /// List of binaries this docker image should include
+    bins: Vec<Bin>,
+    /// Info about the runner image
+    runner: RunnerImage,
+}
+
+/// A binary
+#[derive(Debug, Serialize)]
+struct Bin {
+    /// Path to where the binary should be copied from, on the host system
+    source: String,
+    /// Name of the binary, as exposed to users of the image
+    name: String,
+}
+
+/// Info about the runner image
+#[derive(Debug, Serialize)]
+struct RunnerImage {
+    /// Base image (e.g. some debian distro)
+    image: String,
+    /// Apt deps to install
+    apt_deps: Vec<String>,
+    /// Additional files/dirs to copy into the image (e.g. examples)
+    includes: Vec<Utf8PathBuf>,
+}
+
+/// As of this rewriting this is the latest debian release
+const RUNNER_IMAGE: &str = "debian:bookworm-slim";
+/// What to call the temporary dockerfile we generate
+const DOCKERFILE_NAME: &str = "Dockerfile";
+
+impl DockerInstallerInfo {
+    /// Build the docker image
+    pub fn build(
+        &self,
+        templates: &Templates,
+        dist: &DistGraph,
+        manifest: &DistManifest,
+    ) -> DistResult<()> {
+        let info = self.compute_dockerfile(dist, manifest)?;
+        self.render_dockerfile(templates, &info)?;
+        self.build_and_export_image(dist)?;
+        Ok(())
+    }
+
+    /// Compute the apt runner deps
+    ///
+    /// We want this to be late-bound and to reference DistManifest
+    /// so linkage data can be incorporated.
+    fn apt_runner_deps(&self, dist: &DistGraph, _manifest: &DistManifest) -> Vec<String> {
+        // FIXME: incorporate linkage info to only include Explicit runtime deps
+        // and to also auto-include runtime deps they forgot to mention!
+        // (soft-blocked on doing a sysdeps refactor)
+        let release = dist.release(self.release_idx);
+        let packages: Vec<String> = release
+            .system_dependencies
+            .apt
+            .clone()
+            .into_iter()
+            .filter(|(_, package)| package.0.wanted_for_target(&self.target))
+            .filter(|(_, package)| package.0.stage_wanted(&DependencyKind::Run))
+            .map(|(name, spec)| {
+                if let Some(version) = spec.0.version {
+                    format!("{name}={version}")
+                } else {
+                    name
+                }
+            })
+            .collect();
+        packages
```

Comment:
You can probably just return directly instead of assigning and returning, right?